controller.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2018 Mike Bryant. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package main for the random secret operator
+package main
+
+import (
+	"fmt"
+
+	"crypto/rand"
+
+	"encoding/hex"
+
+	randomsecrets "github.com/mikebryant/random-secret-operator/pkg/apis/randomsecrets/v1"
+	randomsecretsclient "github.com/mikebryant/random-secret-operator/pkg/client/clientset/versioned/typed/randomsecrets/v1"
+	opkit "github.com/rook/operator-kit"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+)
+
+// RandomSecretController represents a controller object for random secret custom resources
+type RandomSecretController struct {
+	context               *opkit.Context
+	randomSecretClientset randomsecretsclient.RandomsecretsV1Interface
+}
+
+// newRandomSecretController create controller for watching random secret custom resources created
+func newRandomSecretController(context *opkit.Context, randomSecretClientset randomsecretsclient.RandomsecretsV1Interface) *RandomSecretController {
+	return &RandomSecretController{
+		context:               context,
+		randomSecretClientset: randomSecretClientset,
+	}
+}
+
+// Watch watches for instances of RandomSecret custom resources and acts on them
```

Comment:
comment on exported method RandomSecretController.StartWatch should be of the form "StartWatch ..."

Reply:
Done :)